examples/database/main.tf

```diff
@@ -0,0 +1,93 @@
+terraform {
+  required_version = "~> 1.6"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 3.108"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = "~> 3.6"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+variable "enable_telemetry" {
+  type        = bool
+  default     = true
+  description = <<DESCRIPTION
+This variable controls whether or not telemetry is enabled for the module.
+For more information see<https://aka.ms/avm/telemetryinfo>.
+If it is set to false, then no telemetry will be collected.
+DESCRIPTION
+}
+
+# This ensures we have unique CAF compliant names for our resources.
+module "naming" {
+  source  = "Azure/naming/azurerm"
+  version = "0.3.0"
+}
+
+# This is required for resource modules
+resource "azurerm_resource_group" "this" {
+  location = "AustraliaEast"
+  name     = module.naming.resource_group.name_unique
+}
+
+resource "random_password" "admin_password" {
+  length           = 16
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+  special          = true
+}
+
+locals {
+  databases = {
+    my_sample_db = {
+      create_mode  = "Default"
+      collation    = "SQL_Latin1_General_CP1_CI_AS"
+      server_id    = module.sql_server.resource.id
```

Comment:
Isn't this causing a circular dependency?

Reply:
Also, why not `resource_id`?

Reply:
Done

Reply:
Still seeing this? I don't understand the logic, seems to be relying on the internals of the module. I would not expect to see the `server_id` field here as the module should supply that. This would only be relevant where we have a submodule for databases.